core/src/com/google/inject/Injector.java

```diff
@@ -269,7 +271,32 @@
    * <p>The returned list does not include elements inherited from a {@link #getParent() parent
    * injector}, should one exist.
    *
+   * <p>The returned list is immutable; it contains only the elements that were present when {@link
+   * #getElements} was invoked. Subsequent calls may return a list with additional elements.
+   *
+   * <p>The returned list does not include data inherited from a {@link #getParent() parent
+   * injector}, should one exist.
+   *
+   * <p>This method is part of the Guice SPI and is intended for use by tools and extensions.
+   *
    * @since 5.0
    */
   List<Element> getElements();
+
+  /**
+   * Returns the injection points created for calls to {@link #getMembersInjector} (either directly
+   * or indirectly, e.g. through {@link #injectMembers}.
+   *
+   * <p>This excludes any injection points from elements (which are accessible from each element via
+   * the SPI), unless {@link #getMembersInjector} or {@link #injectMembers} were also called for the
+   * same key.
+   *
+   * <p>The returned multimap does not include data inherited from a {@link #getParent() parent
+   * injector}, should one exist.
+   *
+   * <p>This method is part of the Guice SPI and is intended for use by tools and extensions.
+   *
+   * @since 5.0
+   */
+  ListMultimap<TypeLiteral<?>, InjectionPoint> getAllMembersInjectorInjectionPoints();
```

Comment:
I think this would be the first time a Guava type was exposed through the public API - is it worth considering something more generic such as `Map<TypeLiteral<?>, List<InjectionPoint>>` to avoid leaking implementation concerns?

Reply:
Excellent point Stuart. We do use c.g.c.base.Optional for use with OptionalBinder, but that's more of "support" for it rather than exposing it. 

Do you think the direct Guava ref will cause issues?  Exposing the type as a multimap does offer some nice conveniences.  I don't quite see it as an impl detail since it's the interface supertype... but if it can cause issues then we can change to bland Map+List.

Reply:
It would only cause issues if the caller wanted to use a different version of Guava to the one that Guice was using behind the API, but that's a relatively obscure scenario. Personally I lean towards JDK interfaces wherever possible, but I don't know how you're planning to use this API and whether that needs the multimap conveniences.

Reply:
Thanks Stuart, we're submitting a change to use Map<K, List<V>>.  We'll merge this as-is and the next PR pushing out changes will change the signature.